data_describe/anomaly_detection/detection.py

```diff
@@ -0,0 +1,548 @@
+    and then projects the data onto a plot for visualization.
+
+    Args:
+        data (DataFrame): The dataframe.
+        target (str, optional): The target column. Defaults to None. If target is None, unsupervised methods are used.
+        date_col (str, optional): The datetime column. If date_col is specified, the data will be treated as timeseries.
+            If the data does not contain datetimes, but contains sequences, set date_col = 'index'.
+        method (str, optional): Select method from the list: ["arima"]. Only "arima" is supported.
+        estimator (optional): Fitted or instantiated estimator with a .predict() and .fit() method. Defaults to None.
+            If estimator is instantiated but not fitted, data_split must be specified.
+        data_split (int, optional): Index to split the data into a train set and a test set. Defaults to None.
+            data_split must be specified if estimator is instantiated but not fitted.
+        n_periods (int, optional): Number of periods for timeseries anomaly detection. Defaults to None.
+        sigma (float, optional): The standard deviation requirement for identifying anomalies. Defaults to 2.
+        compute_backend (str, optional): The compute backend.
+        viz_backend (str, optional): The visualization backend.
+        **kwargs: Keyword arguments.
+
+    Return:
+        AnomalyWidget
+    """
+    # checks if input is dataframe
+    if not _is_dataframe(data):
+        raise ValueError("Data frame required")
+
+    # checks if estimator exists and if it follows the .predict and .fit methods
+    if estimator:
+        if not hasattr(estimator, "predict") and not hasattr(estimator, "fit"):
+            raise AttributeError(
+                "Input model does not contain the 'predict' or 'fit' method."
+            )
+    # TODO(truongc2): Update available methods
+    # ml_methods = {
+    #     "timeseries": ["arima"],
+    # }
+
+    numeric_data = data.select_dtypes("number")
+
+    # ensures date_col is a datetime object and sets as datetimeindex
+    if date_col:
+        if date_col != "index":
+            numeric_data.index = pd.to_datetime(data[date_col])
+            if not numeric_data.index.is_monotonic_increasing:
+                numeric_data.sort_index(inplace=True)
+
+    anomalywidget = _get_compute_backend(compute_backend, numeric_data).compute_anomaly(
+        data=numeric_data,
+        target=target,
+        date_col=date_col,
+        estimator=estimator,
+        n_periods=n_periods,
+        data_split=data_split,
+        method=method,
+        **kwargs,
+    )
+
+    anomalywidget.viz_backend = viz_backend
+    anomalywidget.date_col = date_col
+    anomalywidget.target = target
+    anomalywidget.viz_backend = viz_backend
+    anomalywidget.n_periods = n_periods
+    anomalywidget.sigma = sigma
+    anomalywidget.xlabel = date_col
+    anomalywidget.ylabel = target
```

Comment:
these attributes (or at least most of them) should already be initialized when passed through the compute method